src/ts/components/settingspanelnavigatorbutton.ts

```diff
@@ -0,0 +1,45 @@
+import {Button, ButtonConfig} from './button';
+import {SettingsPanel} from './settingspanel';
+import {SettingsPanelPage} from './settingspanelpage';
+
+/**
+ * Configuration interface for a {@link SettingsPanelNavigatorButton}
+ */
+export interface SettingsPanelNavigatorConfig extends ButtonConfig {
+  /**
+   * Container `SettingsPanel` where the navigation takes place
+   */
+  container: SettingsPanel;
+  /**
+   * Page where the button should navigate to
+   * If empty it will navigate to the root page (not intended to use as navigate back behavior)
+   */
+  targetPage?: SettingsPanelPage;
+}
+
+export class SettingsPanelNavigatorButton extends Button<SettingsPanelNavigatorConfig> {
```

Comment:
Please call them `SettingsPanelPageNavigatorButton` and `SettingsPanelPageNavigatorButtonConfig` (yes its along name but that's the pattern)